src/socket.rs

```diff
@@ -2205,6 +2240,120 @@ impl Socket {
+    pub fn original_dst_ipv6(&self) -> io::Result<SockAddr> {
+        // Safety: `getsockopt` initialises the `SockAddr` for us.
+        unsafe {
+            SockAddr::try_init(|storage, len| {
+                syscall!(getsockopt(
+                    self.as_raw(),
+                    libc::SOL_IPV6,
+                    libc::IP6T_SO_ORIGINAL_DST,
+                    storage.cast(),
+                    len
+                ))
+            })
+        }
+        .map(|(_, addr)| addr)
+    }
+
+    /// Get the value for the `SO_ORIGINAL_DST` option on this socket.
+    /// Only valid for sockets in accepting mode.
+    ///
+    /// Note: if using this function in a proxy context, you must query the
+    /// redirect records for this socket and set them on the outbound socket
+    /// created by your proxy in order for any OS level firewall rules to be
+    /// applied. Read more in the Windows bind and connect redirection
+    /// [documentation](https://learn.microsoft.com/en-us/windows-hardware/drivers/network/using-bind-or-connect-redirection).
+    #[cfg(all(windows, feature = "all"))]
+    #[cfg_attr(docsrs, doc(cfg(all(windows, feature = "all"))))]
+    pub fn original_dst(&self) -> io::Result<SockAddr> {
+        unsafe {
+            SockAddr::try_init(|storage, len| {
+                syscall!(
+                    getsockopt(
+                        self.as_raw(),
+                        SOL_IP as i32,
+                        SO_ORIGINAL_DST as i32,
+                        storage.cast(),
+                        len,
+                    ),
+                    PartialEq::eq,
+                    SOCKET_ERROR
+                )
+            })
+        }
+        .map(|(_, addr)| addr)
+    }
+
+    /// Get the value for the `IP6T_SO_ORIGINAL_DST` option on this socket.
+    /// Only valid for sockets in accepting mode.
+    ///
+    /// Note: if using this function in a proxy context, you must query the
+    /// redirect records for this socket and set them on the outbound socket
+    /// created by your proxy in order for any OS level firewall rules to be
+    /// applied. Read more in the Windows bind and connect redirection
+    /// [documentation](https://learn.microsoft.com/en-us/windows-hardware/drivers/network/using-bind-or-connect-redirection).
+    #[cfg(all(windows, feature = "all"))]
+    #[cfg_attr(docsrs, doc(cfg(all(windows, feature = "all"))))]
+    pub fn original_dst_ipv6(&self) -> io::Result<SockAddr> {
+        unsafe {
+            SockAddr::try_init(|storage, len| {
+                syscall!(
+                    getsockopt(
+                        self.as_raw(),
+                        SOL_IP as i32,
+                        IP6T_SO_ORIGINAL_DST as i32,
+                        storage.cast(),
+                        len,
+                    ),
+                    PartialEq::eq,
+                    SOCKET_ERROR
+                )
+            })
+        }
+        .map(|(_, addr)| addr)
+    }
```

Comment:
These functions should be merged with the Unix version, see all the other functions here.

Reply:
Ah I see, I misunderstood your original instructions. Will fix